### ISSUE TITLE
Build script for AppImage uses outdated 'appimagetool' not supporting future auto-updates of AppImages

### ISSUE THREAD

**Issue by KurtPfeifle** [PowerShell/PowerShell#5879]
### TL;DR: Please update and make use of newer 'appimagetool' to enable automatic updates

The build script 'appimage.sh' uses a self-hosted 'appimagetool' binary which is more than 1 year old. 

https://github.com/PowerShell/PowerShell/blob/2ae61c8ce46de49c6e0feb86ec0f91e4323b3fb4/tools/appimage.sh#L472

This binary generates the final AppImage from the AppDir structure which previous steps of 'appimage.sh' had prepared:

https://github.com/PowerShell/PowerShell/blob/2ae61c8ce46de49c6e0feb86ec0f91e4323b3fb4/tools/appimage.sh#L474

This "old" appimagetool misses a very useful feature supported by more recent releases:

* The most **recent versions of 'appimagetool' can embed an *update string* into the final AppImage**. 

This is very beneficial for users: it lets them trigger the automatic update of an AppImage to its newer version(s). This update uses the very efficient download of only the "binary delta" sections of any new AppImage that is stored at the original hosting location of their previous AppImage -- and the user does not even need to remember where that location was.

A new version of 'appimagetool' would be run with the *`-u | --updateinformation`* like this:

    ./appimagetool-x86_64.AppImage \
          -u "gh-releases-zsync|PowerShell|PowerShell|continuous|powershell-6*x86_64.AppImage.zsync" \
          ./powershell.AppDir

<sup>(For currently available variations of the update info string see their descriptions in the AppImage Spec document: https://github.com/AppImage/AppImageSpec/blob/master/draft.md#update-information)</sup>

---- 

This would, besides generating a *powershell-6.5.4-x86_64.AppImage* in the future, additionally generate an extra, rather small *powershell-6.5.4-x86_64.AppImage.zsync* file. The .zsync file contains the hashes of binary blocks composing the (new) AppImage. It should then be hosted side by side to the real AppImage on the GitHub release page. (For an example, see https://github.com/Subsurface-divelog/subsurface/releases).

When querying for available updates this .zsync file would indicate to clients if or if not updates area available, and would help to determine which "delta blocks" need to be downloaded to the user in order to create the new AppImage version there.

This could be verified by running the AppImage with the *`--appimage-updateinfo`* flag:

    $> ./powershell-6.5.4-x86_64.AppImage --appimage-updateinfo
    gh-releases-zsync|PowerShell|PowerShell|continuous|powershell-6*x86_64.AppImage.zsync

The user would trigger the update process either by a GUI or a CLI utility on her side. The CLI invokation (as *`"appimageupdatetool powershell-6.0.0-x86_64.AppImage"`*) would then check for updates and apply them if available, downloading only a fraction of the size of the full new AppImage. As a result, the user will now have two AppImages: *powershell-6.0.0-x86_64.AppImage* (previous) and *powershell-6.5.4-x86_64.AppImage* (new). She could have used *`--overwrite`* to prevent the previous version to be kept around, though.

You can hardly conceive of a more simple and more elegant way to handle updates from the users' point of view.

----

For a full and working real-life example, how this works when a user wants to check for (or really update to) a new AppImage, see this GitHub wiki article:

* https://github.com/AppImage/AppImageUpdate/wiki/Example:-Step-by-step-walkthrough-with-%60appimageupdatetool%60

**Comment by KurtPfeifle** [comment]
I noticed that the AppImage build has been removed in the 6.0.3 release.

Is this a temporary measure? Will it come back? Or is this a permanent situation?
